### PR TITLE
fixes #4682 - smart variable for a puppet class not showing in UI

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

**Comment by stbenjam** [comment]
@abenari @ohadlevy This change improves how parameters and variable counts are presented to the user, but also now that we have the right counts, the edit view for the existing smart variables loads.

File: db/migrate/20140318153157_fix_puppetclass_counters.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class FixPuppetclassCounters < ActiveRecord::Migration
+  def up 
+    # This is the counter for the total number of params for a given puppet class
+    # across all environments.  It was poorly named.
+    rename_column :puppetclasses, :lookup_keys_count, :global_class_params_count
+
+    # Smart Variables Counts 
+    add_column    :puppetclasses, :lookup_keys_count, :integer, :default => 0 
+    LookupKey.where("puppetclass_id IS NOT NULL").each do |k|
+      Puppetclass.increment_counter(k.puppetclass_id, :lookup_keys)
```
Comment (abenari) [inline db/migrate/20140318153157_fix_puppetclass_counters.rb]: migrate didn't work for me,
after changing to the following syntax it worked fine: 
Puppetclass.increment_counter(:lookup_keys_count, k.puppetclass_id)

Comment (stbenjam) [inline db/migrate/20140318153157_fix_puppetclass_counters.rb]: Oh, good catch, I didn't commit my last change.  This should be reset_counters.

**Comment by abenari** [comment]
Tested and the patch is working, and looks good on the UI. Just need to fix the migration issue.

**Comment by abenari** [comment]
merged as df471c78ec92ab321e18e00d232121cddf746337 thanks @stbenjam

**Pull request closed without merging**